sentry-android-replay/src/main/java/io/sentry/android/replay/capture/BaseCaptureStrategy.kt

```diff
@@ -416,4 +521,53 @@ internal abstract class BaseCaptureStrategy(
             else -> null
         }
     }
+
+    private inline fun <T> persistableAtomicNullable(
+        initialValue: T? = null,
+        propertyName: String? = null,
+        crossinline onChange: (propertyName: String?, oldValue: T?, newValue: T?) -> Unit = { _, _, newValue ->
+            propertyName ?: error("Can't persist value without a property name")
+
+            if (options.mainThreadChecker.isMainThread) {
+                persistingExecutor.submit {
+                    cache?.persistSegmentValues(propertyName, newValue.toString())
+                }
+            } else {
+                cache?.persistSegmentValues(propertyName, newValue.toString())
+            }
+        }
+    ): ReadWriteProperty<Any?, T?> =
+        object : ReadWriteProperty<Any?, T?> {
+            private val value = AtomicReference(initialValue)
+
+            init {
+                onChange(propertyName, initialValue, initialValue)
+            }
+
+            override fun getValue(thisRef: Any?, property: KProperty<*>): T? = value.get()
+
+            override fun setValue(thisRef: Any?, property: KProperty<*>, value: T?) {
+                val oldValue = this.value.getAndSet(value)
+                if (oldValue != value) {
+                    onChange(propertyName, oldValue, value)
+                }
+            }
+        }
+
+    private inline fun <T> persistableAtomic(
+        initialValue: T? = null,
+        propertyName: String? = null,
+        crossinline onChange: (propertyName: String?, oldValue: T?, newValue: T?) -> Unit = { _, _, newValue ->
+            propertyName ?: error("Can't persist value without a property name")
+
+            if (options.mainThreadChecker.isMainThread) {
+                persistingExecutor.submit {
+                    cache?.persistSegmentValues(propertyName, newValue.toString())
+                }
+            } else {
+                cache?.persistSegmentValues(propertyName, newValue.toString())
+            }
```

Comment:
this pattern is repeated several times
what about adding an extension to the executor `submitOrNow` or something like that (or a custom Executor class)?

Reply:
yeah, already done here in the other stacked PR: https://github.com/getsentry/sentry-java/pull/3609/files#diff-2f6abcfaf76542e5e58a9fbd4458b23ca658763d9bc0f989abfdeeee5c5898ddR334